### DISCUSSION TITLE
Процесс редактирования руководств по реализации FHIR на сайте Fhir.ru

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#250] [Минимальный набор артефактов]
### Общие положения
Руководства по реализации FHIR в РФ публикуются с помощью стандартного FHIR Publisher и размещаются на страницах вида rucore.fhir.ru, rulab.fhir.ru и т.д.
Артефакты для сборки руководства публикуются в виде текстов [на языке fsh](http://build.fhir.org/ig/HL7/fhir-shorthand/) на страницах сайта Fhir.ru

### Рекомендации по оформлению артефактов
Методические рекомендации и соглашения по оформлению FHIR-артефактов, принятые в рамках fhir.ru, приведены в методических рекомендациях (здесь будет ссылка).

### Расположение страниц с артефактами на сайте Fhir.ru
Рекомендуется в корне сайта располагать заголовочную страницу руководства, описать на ней предназначение и взаимосвязи с другими руководствами.
Для прикладного руководства рекомендуется располагать страницы с артефактами в следующей структуре:
- профили ресурсов
- профили типов данных
- расширения
- наборы значений
- системы кодирования
- системы именования

### Право редактирования страниц сайта
В настоящее время право на редактирование распространяется на все страницы сайта.
Решение о предоставлении права на редактирование принимается на встречах сообщества. 
Право выдает администратор сайта (требуется учетная запись google).

### Процесс редактирования руководства
- Редактор вносит правки на странице с артефактом
- - Редактор заходит на страницу с артефактом, нажимает кнопку "редактировать", при необходимости авторизуется (в текущей версии сайта рекомендуется авторизоваться на стартовой странице сайта).
- - В открывшейся панели редактирования вносит необходимые правки
- - при этом, для fsh-текста использует тег :profile fsh/
- - <img width="1390" alt="image" src="https://github.com/fhir-ru/core/assets/62525329/33b59f1f-90f3-482e-9a90-fcdcaafc61cb">
- - Нажимает кнопку "сохранить"
- Редактор просит администратора сайта пересобрать руководства (планируется поддержать автоматическую сборку руководств по команде редактора)

**Comment by Vrodvi** [comment]
# Замечание 1.

> Артефакты для сборки руководства публикуются в виде текстов [на языке fsh](http://build.fhir.org/ig/HL7/fhir-shorthand/) на страницах сайта Fhir.ru

1.1. Желательно добавить небольшую инструкцию о формате разметки Zendoc, с помощью которой автор может вставить FSH-описания.

 1.2. Желательно добавить пояснения о рекомендуемом наполнении FSH-описаний - может ли в них описываться сразу несколько артефактов (к примеру, профиль, расширение, набор значений к нему и соответствующий справочник)? Или на каждый тип артефакта строго должен быть на отдельной странице в соответствующем разделе описан? Инструмент SUSHI, к примеру даёт полную свободу в плане авторства артефактов, так что, в пределе, можно вообще все артефакты описать на одной странице. 
Какой подход рекомендуется при работе через Zendoc?
Также, очевидно, что большинство расширений, к примеру, будут относится строго к одному профилю - имеет ли смысл описывать их на отдельной странице или лучше описать их в профиле для которого они предназначены, а на странице "Расширения" будет просто таблица со ссылками на все места, где описаны расширения?

1.3. Желательно добавить пояснения о разметке описательной части артефактов - каким образом следует оформлять разметку в Zendoc, чтобы эта описательная часть переносилась в стандартное руководство по реализации в виде введения и подвала (header, footer), см. соответствующее обсуждение тут - https://github.com/fhir-ru/core/discussions/235#discussioncomment-7764147

**Reply by eugenekogan-spb** [reply]
1.1. добавлено
1.2.1. Ранее мы решали, что на странице профиля будет профиль и примеры, на странице системы кодирования будет NamingSystem и CodeSystem и т.д. Я бы предложил оставить эти пояснения в рекомендациях по описанию артефактов. Здесь можно дать ссылки.
1.2.2. Поддерживаю идею о включении расширений на страничку с расширяемым артефактом. Внес изменения в методические рекомендации: https://github.com/fhir-ru/core/discussions/246. TODO: перенести на сайте и сделать автоматически генерируемую табличку на странице Расширения.
1.3. Ждем реализации.
